application/F3DOptionsParser.cxx

```diff
@@ -374,11 +393,48 @@ void ConfigurationOptions::GetOptions(F3DAppOptions& appOptions, f3d::options& o
     cxxOptions.positional_help("file1 file2 ...");
     cxxOptions.parse_positional({ "input" });
     cxxOptions.show_positional_help();
+    cxxOptions.allow_unrecognised_options();
 
     if (parseCommandLine)
     {
       auto result = cxxOptions.parse(this->Argc, this->Argv);
 
+      auto unmatched = result.unmatched();
+
+      if (unmatched.size() > 0)
+      {
+        constexpr int distanceThreshold = 5;
+
+        for (std::string unknownOption : unmatched)
+        {
+          // check if it's a long option
+          if (unknownOption[0] == '-' && unknownOption[1] == '-')
+          {
+            // remove trailing '--'
+            unknownOption = unknownOption.substr(2);
+
+            auto it = std::find_if(unknownOption.rbegin(), unknownOption.rend(),
+              [](unsigned char ch) { return ch == '='; });
+
+            // remove everything after the character '='
+            if (it != unknownOption.rend())
+            {
+              unknownOption.erase(it.base() - 1, unknownOption.end());
+            }
+
+            f3d::log::error("Unknown option --", unknownOption);
+
+            auto [name, dist] = this->GetClosestOption(unknownOption);
+
+            if (dist < distanceThreshold)
```

Comment:
is it worth checking ? Better to provide a wrong suggestion than no suggestion at all.